code/core/src/main/java/com/adobe/marketing/mobile/internal/util/StringExtensions.kt

```diff
@@ -0,0 +1,50 @@
+/*
+  Copyright 2024 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+*/
+
+package com.adobe.marketing.mobile.internal.util
+
+/**
+ * Computes the signed 2's complement FNV-1a 32-bit hash for the given string input.
+ *
+ * <p>https://en.wikipedia.org/wiki/Fowler%E2%80%93Noll%E2%80%93Vo_hash_function Online
+ * validator - https://md5calc.com/hash/fnv1a32?str=
+ *
+ * @param input The string to be hashed.
+ * @return A 32-bit signed integer representing the FNV-1a hash of the input string.
+ */
+private fun computeFnv1aHash(input: String?): Int {
+    val ALL_BITS_ENABLED = 0xFF
+    val PRIME = 0x1000193 // 16777619 as hex
+    val OFFSET = -0x7ee3623b // 2166136261 as hex
+
+    if (input == null) {
+        return 0
+    }
```

Comment:
Will it be simpler if we combine null+empty input validation and move it earlier -   
```
    if (input.isNullOrEmpty()) return 0

    val ALL_BITS_ENABLED = 0xFF
    val PRIME = 0x1000193 // 16777619 as hex
    val OFFSET = -0x7ee3623b // 2166136261 as hex
    var hash = OFFSET
    
    for(...) { ...}
    

```

Reply:
We don't return zero if the string is empty. The hashes are different based on length.

Reply:
I also realized that we should have simply kept var hash = OFFSET, as our implementation diverges from the original algorithm when generating hashes for strings containing only empty characters. However, since this is a corner case, I have decided not to modify the implementation and will leave it as it currently exists in our SDK.